src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,92 @@
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  deleteTodo: (id: number) => void;
+  isProcessing: Todo | null;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  tempTodo,
+  deleteTodo,
+  isProcessing,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map((todo) => (
+        <div
+          key={todo.id}
+          data-cy="Todo"
+          className={classNames('todo', {
+            completed: todo.completed,
+          })}
+        >
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+            />
+          </label>
+
+          <span data-cy="TodoTitle" className="todo__title">
+            {todo.title}
+          </span>
+
+          <button
+            type="button"
+            className="todo__remove"
+            data-cy="TodoDelete"
+            onClick={() => deleteTodo(todo.id)}
+          >
+            ×
+          </button>
+
+          <div
+            data-cy="TodoLoader"
+            className={classNames('modal overlay', {
+              'is-active': isProcessing && isProcessing.id === todo.id,
+            })}
+          >
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
+        </div>
+      ))}
+
+      {!!tempTodo && (
+        <div
+          key={tempTodo.id}
+          data-cy="Todo"
+          className={classNames('todo', {
+            completed: tempTodo.completed,
+          })}
+        >
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+            />
+          </label>
+
+          <span data-cy="TodoTitle" className="todo__title">
+            {tempTodo.title}
+          </span>
+
+          <button type="button" className="todo__remove" data-cy="TodoDelete">
+            ×
+          </button>
+
+          <div data-cy="TodoLoader" className="modal overlay is-active">
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
+        </div>
```

Comment:
This part is very similar to the part above that is used to render normal `todos`, so it's worth bringing it out as a separate `TodoItem` component and reusing it.